File: intermediate-package/writing-integ-tests-for-spring-web-apps/rest-api/spring-boot/src/test/java/com/testwithspring/intermediate/web/WebTestConfig.java
package com.testwithspring.intermediate.web;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.SerializationFeature;
import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
import org.springframework.web.servlet.LocaleResolver;
import org.springframework.web.servlet.i18n.FixedLocaleResolver;

import java.util.Locale;

/**
 * This class provides static factory methods that are used
 * to create the Spring MVC infrastructure components that
 * we use when we are writing unit tests for Spring MVC
 * controllers.
 */
public final class WebTestConfig {

    public static final Locale LOCALE = Locale.ENGLISH;

    /**
     * Prevents instantiation.
     */
    private WebTestConfig() {}

    /**
     * This method creates a locale resolver that always returns {@code Locale.ENGLISH}
     * @return
     */
    public static LocaleResolver fixedLocaleResolver() {
        return new FixedLocaleResolver(LOCALE);
    }

    /**
     *
     * @return
     */
    public static MappingJackson2HttpMessageConverter objectMapperHttpMessageConverter() {
        MappingJackson2HttpMessageConverter converter = new MappingJackson2HttpMessageConverter();
        converter.setObjectMapper(objectMapper());
        return converter;
    }

    public static ObjectMapper objectMapper() {
        return new ObjectMapper()
                .disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)
                .registerModule(new Jdk8Module());
    }
}

File: master-package/bonus-lessons-2020/integration-tests-rest-api/src/test/kotlin/com/testwithspring/master/web/WebTestConfig.kt
package com.testwithspring.master.web

import com.fasterxml.jackson.datatype.jdk8.Jdk8Module
import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.databind.SerializationFeature
import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter
import org.springframework.web.servlet.LocaleResolver
import org.springframework.web.servlet.i18n.FixedLocaleResolver
import java.util.Locale
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
import org.springframework.http.converter.json.Jackson2ObjectMapperBuilder

/**
 * This class provides static factory methods that are used
 * to create the Spring MVC infrastructure components that
 * we use when we are writing unit tests for Spring MVC
 * controllers.
 */
class WebTestConfig private constructor () {

    companion object {
        private val LOCALE = Locale.ENGLISH

        /**
         * This method creates a locale resolver that always returns `Locale.ENGLISH`
         * @return
         */
        fun fixedLocaleResolver(): LocaleResolver {
            return FixedLocaleResolver(LOCALE)
        }

        /**
         * Creates a new {@link MappingJackson2HttpMessageConverter} object and configures
         * the used object mapper.
         * @return
         */
        fun objectMapperHttpMessageConverter(): MappingJackson2HttpMessageConverter {
            val converter = MappingJackson2HttpMessageConverter()
            converter.objectMapper = objectMapper()
            return converter
        }

        /**
         * Creates a new {@link ObjectMapper} object.
         * @return
         */
        fun objectMapper(): ObjectMapper {
            return Jackson2ObjectMapperBuilder()
                    .featuresToDisable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)
                    .featuresToEnable(SerializationFeature.WRITE_DATES_WITH_ZONE_ID)
                    .modulesToInstall(
                            Jdk8Module(),
                            JavaTimeModule()
                    )
                    .build()
        }
    }
}